lib/stdlib/src/supervisor.erl

```diff
@@ -334,35 +321,34 @@ init_dynamic(_State, StartSpec) ->
 
 %%-----------------------------------------------------------------
 %% Func: start_children/2
-%% Args: Children = [child_rec()] in start order
+%% Args: Children = children() % Ids in start order
 %%       SupName = {local, atom()} | {global, atom()} | {pid(), Mod}
-%% Purpose: Start all children.  The new list contains #child's
+%% Purpose: Start all children.  The new map contains #child's
 %%          with pids.
 %% Returns: {ok, NChildren} | {error, NChildren, Reason}
-%%          NChildren = [child_rec()] in termination order (reversed
-%%                        start order)
+%%          NChildren = children() % Ids in termination order
+%%                                   (reversed start order)
 %%-----------------------------------------------------------------
-start_children(Children, SupName) -> start_children(Children, [], SupName).
-
-start_children([Child|Chs], NChildren, SupName) ->
-    case do_start_child(SupName, Child) of
-	{ok, undefined} when Child#child.restart_type =:= temporary ->
-	    start_children(Chs, NChildren, SupName);
-	{ok, Pid} ->
-	    start_children(Chs, [Child#child{pid = Pid}|NChildren], SupName);
-	{ok, Pid, _Extra} ->
-	    start_children(Chs, [Child#child{pid = Pid}|NChildren], SupName);
-	{error, Reason} ->
-	    report_error(start_error, Reason, Child, SupName),
-	    {error, lists:reverse(Chs) ++ [Child | NChildren],
-	     {failed_to_start_child,Child#child.name,Reason}}
-    end;
-start_children([], NChildren, _SupName) ->
-    {ok, NChildren}.
+start_children(Children, SupName) ->
+    Start =
+        fun(Id,Child) ->
+                case do_start_child(SupName, Child) of
```

Comment:
Apologies for a comment on the style but I believe this is indented too much.

Reply:
That's actually a bit strange - it's the indentation done by the erlang emacs mode on funs... I will check why...

Reply:
Apparently, this is a correct indentation - it is like this to cover for a combination of multiple fun clauses in combination with newline before guards.. e.g. 
```
fun(X)
      when X > 0 ->
        positive;
   (X)
      when X < 0 ->
        negative;
   (0) ->
        zero
end
```
... and the main problem lies in the second clause, where it would be very strange if one indentation level was removed:
```
fun(X)
  when X > 0 ->
    positive;
   (X)
  when X < 0 ->
    negative;
   (0) ->
    zero
end
```